### ISSUE TITLE
[Bug - next-pwa]: sw-entry break service worker skipWaiting behavior

### ISSUE THREAD

**Issue by aNyMoRe0505** [DuCanhGH/next-pwa#68]
### Provide environment information

next: 13.4.12
@ducanh2912/next-pwa: 9.4.0

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code help us address them faster

https://github.com/aNyMoRe0505/skip-waiting-bug-demo

### To reproduce

1. clone this [repo](https://github.com/aNyMoRe0505/skip-waiting-bug-demo)
2. npm ci && npm run build && npm run start
3. check http://localhost:3000/   dev console application
4. this is the first time install service worker, everything work correctly
![截圖 2023-07-28 13 05 19](https://github.com/DuCanhGH/next-pwa/assets/9814815/57c78bb1-b4c7-4f31-8286-027a3731209a)
5. stop the server, update the console.log text in `worker/index.js` (for browser know there is a new version worker)
6. npm run build && npm run start then reload http://localhost:3000/  check dev console application
![截圖 2023-07-28 13 10 20](https://github.com/DuCanhGH/next-pwa/assets/9814815/e3b973c9-2e93-4bc1-a8ff-a9870b20b247)

you will find the worker state is `waiting to activate` even [the default skipWaiting config](https://github.com/DuCanhGH/next-pwa/blob/master/packages/next-pwa/src/resolve-workbox-plugin.ts#L55) is true

if you reload the page, you will find there is a pending request (never come back), you need to reload again let the site work properly

### Describe the bug

After debugging, I found the reason is I have an `otherPage` link in [index.js](https://github.com/aNyMoRe0505/skip-waiting-bug-demo/blob/main/pages/index.js#L21)

since [otherPage](https://github.com/aNyMoRe0505/skip-waiting-bug-demo/blob/main/pages/other.js#L5) use getStaticProps, next.js link will [prefetch the data](https://nextjs.org/docs/pages/building-your-application/routing/linking-and-navigating)

and [sw-entry.js](https://github.com/DuCanhGH/next-pwa/blob/master/packages/next-pwa/src/sw-entry.ts#L64-L75) will add prefetch json data to cache when workbox installed event trigger

```js
  window.workbox.addEventListener("installed", async () => {
    const nextDataCache = await caches.open("next-data");
    window.performance.getEntriesByType("resource").forEach((entry) => {
      const entryName = entry.name;
      if (
        entryName.startsWith(`${window.location.origin}/_next/data/`) &&
        entryName.endsWith(".json")
      ) {
        nextDataCache.add(entryName);
      }
    });
  });
```
After my testing, this will break the skipWaiting behavior, I'm not sure if this is a good practice (handle cache after service worker install, from my understanding, we should handle cache in [service worker install event](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerGlobalScope/install_event))

and [sw-entry.js](https://github.com/DuCanhGH/next-pwa/blob/master/packages/next-pwa/src/sw-entry.ts#L46-L62) will cache start url when workbox installed event trigger also since it only cache start url when `isUpdate` is false, I'm not sure if this will break skipWaiting behavior also.

```js
  if (__PWA_START_URL__) {
    window.workbox.addEventListener("installed", async ({ isUpdate }) => {
      if (!isUpdate) {
        const cache = await caches.open("start-url");
        const response = await fetch(__PWA_START_URL__);
        let _response = response;
        if (response.redirected) {
          _response = new Response(response.body, {
            status: 200,
            statusText: "OK",
            headers: response.headers,
          });
        }
        await cache.put(__PWA_START_URL__, _response);
      }
    });
  }
```

I think we can remove the [logic](https://github.com/DuCanhGH/next-pwa/blob/master/packages/next-pwa/src/sw-entry.ts#L64-L75), since [default cache config](https://github.com/DuCanhGH/next-pwa/blob/master/packages/next-pwa/src/cache.ts#L117-L127) has `next-data` already

But I don't have a comprehensive understanding of the entire project and this could be a breaking-change I think (?)

or add an experimental flag let user disable the logic

thank you very much !

### Expected behavior

new service worker should skipWaiting and turn to activated when the option `skipWaiting` is true

**Comment by DuCanhGH** [comment]
Can you trying changing `"installed"` to `"activated"`? That should fix the issue. Most likely the original owner of the package added this logic so as to implement something similar to precaching for these JSON files, which I'm not sure whether we could determine a list of them that we can add to `additionalManifestEntries`  at build time.

**Comment by aNyMoRe0505** [comment]
@DuCanhGH 
yes, changing `installed` to `activated` can fix this issue
but I'm not sure if this is a proper way

**Comment by DuCanhGH** [comment]
@aNyMoRe0505 yeah I'm not exactly sure either, but I don't think we can move this whole caching stuff somewhere else. Perhaps we can consider removing that altogether, but let me think about it.

**Comment by DuCanhGH** [comment]
Let's just assume `next-pwa` won't just straight up break by removing these event listeners 💀 If anything else I will just deploy a fix :)
